### PR TITLE
Switch issue templates to yaml forms

### PR DESCRIPTION
**Pull request by denisonbarbosa**
The markdown templates were weird to navigate since they just dumped the text in the `issue` body. This makes the process confusing for users and does not achieve what we were aiming for, since (in most cases) the users would delete everything and write the issue on their own.

GitHub supports forms for its issue templates and those can be written in `yaml`, which is a bit easier to navigate. This PR adds forms for bug reports and feature requests so that we can better guide the user to provide all the necessary information we need from the start.

### CODE REVIEW

File: project-repo/templates/common/.github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+        Your additional work here is greatly appreciated and will help us respond as quickly as possible. For general
+        support or usage questions, use [Ubuntu Discourse](https://discourse.ubuntu.com/c/desktop/8).
+
+        By submitting an Issue to this repository, you agree to the terms within the
+        [Ubuntu Code of Conduct](https://ubuntu.com/community/code-of-conduct).
+
+        Be careful with sensitive information and security vulnerabilities. In order to report bugs that could contain
+        sensitive information, use [Launchpad](https://bugs.launchpad.net/ubuntu/+source/PROJECT_TODO/+filebug) instead.
+        On Ubuntu machines, you can use `ubuntu-bug PROJECT_TODO` to collect relevant information.
```
Comment (didrocks) [inline project-repo/templates/common/.github/ISSUE_TEMPLATE/bug_report.yml:20]: I think the previous example was a little bit more clear in that regard. This needs to be at the top to warn the user. Also, it had a disctinct line to say it’s important:
```
>**Please do not report security vulnerabilities here**
>
> Please use [launchpad private bugs](https://bugs.launchpad.net/ubuntu/+source/PROJECT_TODO/+filebug) which is monitored by our security team. On ubuntu machine, it’s best to use `ubuntu-bug PROJECT_TODO` to collect relevant information.
> FIXME: snap?
```

So, with your new better wording, to you mind moving that back to the top?

File: project-repo/templates/common/.github/ISSUE_TEMPLATE/bug_report.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      label: Environment
+      description: |
+        Provide details about the environment you experienced the issue in.
+        * Distribution and version
+        * PROJECT_TODO version
+        * Desktop environment
+        * Hardware
+        * etc.
+
+        Ubuntu users can run `ubuntu-bug PROJECT_TODO --save=/tmp/report` and paste the file content bellow.
```
Comment (didrocks) [inline project-repo/templates/common/.github/ISSUE_TEMPLATE/bug_report.yml:61]: typo: below*.

I would start the part with ubuntu user, as it’s going to collect those information above. So it’s either one or the other.

Also, the "below" had triple markdown backquote marked as raw type: the goal is that content of files and other pecial symbols are not marshalled or converted by mistake to markdown, please reuse this.

File: project-repo/templates/common/.github/ISSUE_TEMPLATE/bug_report.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        Please look at our [Troubleshoot guidelines](#) and provide the debug logs for PROJECT_TODO.
+
+        Remember to redact any sensitive information from them.
+    validations:
+      required: true
+  - type: checkboxes
+    attributes:
+      label: Is there any sensitive information in the logs?
+      options:
+        - label: I have redacted any sensitive information from the logs
```
Comment (didrocks) [inline project-repo/templates/common/.github/ISSUE_TEMPLATE/bug_report.yml:79]: what does it do? Can it prevent publishing it? I don’t see information or automation about this.

File: project-repo/templates/common/.github/ISSUE_TEMPLATE/feature_request.yml [hunk 4]
```diff
@@ -0,0 +1,83 @@
+name: Request a feature
+description: Request a feature
```
Comment (didrocks) [inline project-repo/templates/common/.github/ISSUE_TEMPLATE/feature_request.yml:2]: Shouldn’t that be `Suggest new functionality for this project.` as in the previous scaffolding?

File: project-repo/templates/common/.github/ISSUE_TEMPLATE/feature_request.yml [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+        By submitting an Issue to this repository, you agree to the terms within the
+        [Ubuntu Code of Conduct](https://ubuntu.com/community/code-of-conduct).
+
+        Be careful with sensitive information and security vulnerabilities. In order to report bugs that could contain
+        sensitive information, use [Launchpad](https://bugs.launchpad.net/ubuntu/+source/PROJECT_TODO/+filebug) instead.
+        On Ubuntu machines, you can use `ubuntu-bug PROJECT_TODO` to collect relevant information.
+
+
+        FIXME: snap?
```
Comment (didrocks) [inline project-repo/templates/common/.github/ISSUE_TEMPLATE/feature_request.yml:23]: ditto other file

**Review by didrocks — CHANGES_REQUESTED** [review]
It’s great to see this being a little bit more formalized and we should work on that direction.

I think some renaming/reformatting felt of through the crack, do you mind looking at them?

Also, some parts were redacted in such a way to reduce confusion from the reporter, separating ubuntu systems for non ubuntu systems, highlighting security vulnerabilities…
Please have a look at them.

**Comment by tim-hm** [comment]
Thanks for this @denisonbarbosa one bit from me ... can you follow the commit style as well :pray:

File: project-repo/templates/common/.github/ISSUE_TEMPLATE/bug_report.yml (hunk 1 above)
Comment (denisonbarbosa) [reply project-repo/templates/common/.github/ISSUE_TEMPLATE/bug_report.yml:20]: There's also an argument to be made that we should thank the user for sparing the time to report the issue, no?

I agree on the marking, though. It's better to be more explicit.

File: project-repo/templates/common/.github/ISSUE_TEMPLATE/bug_report.yml [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        Ubuntu users can run `ubuntu-bug PROJECT_TODO --save=/tmp/report` and paste the file content bellow.
+
+        FIXME: snap?
+    validations:
+      required: true
+  - type: textarea
+    attributes:
+      label: Relevant information
+      description: >
+        Please look at our [Troubleshoot guidelines](#) and provide the debug logs for PROJECT_TODO.
```
[the remark this answers is not in the document]
Comment (denisonbarbosa) [reply project-repo/templates/common/.github/ISSUE_TEMPLATE/bug_report.yml:70]: Yeah, that was the idea. Not necessarily to the contributing.md file, though. In `aad-auth`, this is a link to our Logging guide to inform the user how to enable debug logs for the project.

File: project-repo/templates/common/.github/ISSUE_TEMPLATE/bug_report.yml (hunk 3 above)
Comment (denisonbarbosa) [reply project-repo/templates/common/.github/ISSUE_TEMPLATE/bug_report.yml:79]: This is supposed to show a checkbox (and the ticking is required) to reinforce the user about the need to redact sensitive information. It does not prevent the user from ticking it and submitting the issue anyway, but it's another layer at least.

File: project-repo/templates/common/.github/ISSUE_TEMPLATE/bug_report.yml (hunk 2 above)
Comment (denisonbarbosa) [reply project-repo/templates/common/.github/ISSUE_TEMPLATE/bug_report.yml:61]: I'm pretty there's a way to mark it as plain text so nothing gets formatted. I'll look into it.

File: project-repo/templates/common/.github/ISSUE_TEMPLATE/bug_report.yml (hunk 1 above)
Comment (didrocks) [reply project-repo/templates/common/.github/ISSUE_TEMPLATE/bug_report.yml:22]: Right, but the main argument is "the most important information on top". I think it’s fine to tell the user "stop here, please read this, ok to go on, then Thanks for reporting this bug and blablabla".

This is why the previous template had that, and it’s followed up in many ticketting system (launchpad, track, gitlab…)

File: project-repo/templates/common/.github/ISSUE_TEMPLATE/bug_report.yml [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+        * Application settings (redact/remove sensitive information)
+        * Screenshots
+    validations:
+      required: true
+  - type: textarea
+    attributes:
+      label: Environment
+      description: |
+        Provide details about the environment you experienced the issue in.
+        * Distribution and version
```
[the remark this answers is not in the document]
Comment (didrocks) [reply project-repo/templates/common/.github/ISSUE_TEMPLATE/bug_report.yml]: There is not a "please run TODO" as in my example above and initial template (and I think the project version should be on top, as in the previous template, as it’s the most important info to a particular project)

File: project-repo/templates/common/.github/ISSUE_TEMPLATE/bug_report.yml (hunk 2 above)
Comment (didrocks) [reply project-repo/templates/common/.github/ISSUE_TEMPLATE/bug_report.yml]: tell me if you find anything without the markup

File: project-repo/templates/common/.github/ISSUE_TEMPLATE/bug_report.yml (hunk 3 above)
Comment (didrocks) [reply project-repo/templates/common/.github/ISSUE_TEMPLATE/bug_report.yml:82]: ok, I agree that’s better than not having it.

File: project-repo/templates/common/.github/ISSUE_TEMPLATE/bug_report.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (didrocks) [reply project-repo/templates/common/.github/ISSUE_TEMPLATE/bug_report.yml]: So, if we reference something by default, this should maybe be a separate markdown file that we add to the template? Either it’s a section in CONTRIBUTING.md and we reference it, or an additional file we had. The goal is really the template to help bootstrapping a project and have the least amount to do. If we want to then link to anything else, people still can, but the default is guided.

**Review by didrocks — COMMENTED** [review]
ok, I will let you the raw text to figure out if you think you can skip the triple backquote markup.

Answering to your feedback and marking some elements as solved in between.

File: project-repo/templates/common/.github/ISSUE_TEMPLATE/bug_report.yml (hunk 2 above)
Comment (denisonbarbosa) [reply project-repo/templates/common/.github/ISSUE_TEMPLATE/bug_report.yml]: I already switched it. You can add a "parser" option to the validator section and then GitHub will recognize what's typed in there as if it was written in the specified language

File: README.md [hunk 8]
```diff
@@ -1,3 +1,7 @@
 # Desktop Team Engineering Resources
 
 A collection of resources to standardise the Team's engineering processes and templates.
+
+## Troubleshooting
```
Comment (didrocks) [inline README.md:5]: I think we rather want to add this to the Troubleshooting section of the project-repo common template's README, no? (as one of the last items)

File: project-repo/templates/common/.github/ISSUE_TEMPLATE/feature_request.yml [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+        TODO: how to collect them
+        ```
+
+        FIXME: snap?
+  - type: textarea
+    attributes:
+      label: Relevant information
+      description: >
+        Please look at our
+        [PROJECT_TODO Troubleshooting guide](../#troubleshooting)
```
Comment (didrocks) [inline project-repo/templates/common/.github/ISSUE_TEMPLATE/feature_request.yml:86]: Have you tested this with a real bug? I think the github UI renders the page, and so, has the readme attached and the link then may work, but just double checking :)

**Review by didrocks — CHANGES_REQUESTED** [review]
A couple of comments on the new changes, and then, I think we’ll good for +1 :)

File: README.md (hunk 8 above)
Comment (denisonbarbosa) [reply README.md:5]: Oops... Added to the wrong README, my bad.

File: project-repo/templates/common/.github/ISSUE_TEMPLATE/feature_request.yml (hunk 9 above)
Comment (denisonbarbosa) [reply project-repo/templates/common/.github/ISSUE_TEMPLATE/feature_request.yml:86]: I tested it, yes. If we use the `../` it points to the homepage of the repo (you can check on the repo that I've shared with you) and then the `#troubleshooting` directs you to the `##Troubleshooting` item

**Review by didrocks — APPROVED** [review]
Excellent! This is an excellent enhancement for our projects, thanks a lot!

**Review by tim-hm — APPROVED** [review]

**Pull request merged**